storage3/types.py

```diff
@@ -23,12 +26,6 @@ class BaseBucket:
     allowed_mime_types: Optional[list[str]]
     type: Optional[str] = None
 
-    def __post_init__(self) -> None:
-        # created_at and updated_at are returned by the API as ISO timestamps
-        # so we convert them to datetime objects
-        self.created_at = dateutil.parser.isoparse(self.created_at)  # type: ignore
-        self.updated_at = dateutil.parser.isoparse(self.updated_at)  # type: ignore
-
```

Comment:
does this needs to be ported to pydantic also?

Reply:
No, [pydantic has native support](https://docs.pydantic.dev/2.0/usage/types/datetime/) for `datetime` parsing just by declaring the field type accordingly, including ISO8601 which is what we're using here. This additional parsing step shouldn't be necessary.